ksmt-symfpu/src/test/java/io/ksmt/symfpu/operations/FpToBvTransformerTest.kt

```diff
@@ -0,0 +1,689 @@
+            mkBvToFpExprNoSimplify(fp32Sort, defaultRounding(), a.cast(), false),
+            mapOf("a" to a),
+        )
+    }
+
+
+    @Test
+    fun testFpFromBvExpr() = with(createContext()) {
+        val sign by mkBv1Sort()
+        val e by mkBv16Sort()
+        val sig by mkBv16Sort()
+        testFpExpr(
+            mkFpFromBvExpr(sign.cast(), e.cast(), sig.cast()),
+            mapOf("sign" to sign, "e" to e, "sig" to sig),
+        )
+    }
+
+    @EnabledIfEnvironmentVariable(
+        named = "runLongSymFPUTests",
+        matches = "true",
+    )
+    @Test
+    fun testFpToFpDownExpr() = with(createContext()) {
+        val a by mkFp128Sort()
+        testFpExpr(
+            mkFpToFpExpr(mkFp16Sort(), defaultRounding(), a),
+            mapOf("a" to a),
+        )
+    }
+
+    @Test
+    fun testIteExpr() = with(createContext()) {
+        val a by mkFp32Sort()
+        val b by mkFp32Sort()
+        testFpExpr(
+            mkIteNoSimplify(trueExpr, a, b),
+            mapOf("a" to a),
+        )
+    }
+
+    @Test
+    fun testBvBoolFormulaExpr() = withContextAndFp64Variables { a, b ->
+        val sort = fp64Sort
+        val zero = mkFpZero(false, sort)
+        mkFpZero(true, sort)
+
+        val asq = mkFpMulExpr(mkFpRoundingModeExpr(KFpRoundingMode.RoundNearestTiesToEven), a, a)
+
+        testFpExpr(
+            mkFpLessExpr(asq, zero),
+            mapOf("a" to a, "b" to b)
+        )
+    }
+
+    private fun <T : KSort> KContext.checkTransformer(
+        transformer: FpToBvTransformer,
+        solver: KSolver<*>,
+        exprToTransform: KExpr<T>,
+        printVars: Map<String, KApp<*, *>>,
+        extraAssert: ((KExpr<T>, KExpr<T>) -> KExpr<KBoolSort>),
+    ) {
+
+        val applied = transformer.apply(exprToTransform)
+        val transformedExpr: KExpr<T> = ((applied as? UnpackedFp<*>)?.toFp() ?: applied).cast()
+
+        val testTransformer = TestTransformerUseBvs(this, transformer.mapFpToUnpackedFp)
+        val toCompare = testTransformer.apply(exprToTransform)
+
+
+        solver.assert(!mkEqNoSimplify(transformedExpr, toCompare.cast()))
+
+        val status =
+            solver.checkWithAssumptions(
+                listOf(testTransformer.apply(extraAssert(transformedExpr, toCompare))),
+                timeout = 2.seconds
+            )
+        if (status == KSolverStatus.SAT) {
+            val model = solver.model()
+            val transformed = model.eval(transformedExpr)
+            val baseExpr = model.eval(toCompare)
+
+            println("transformed: ${unpackedString(transformed, model)}")
+            println("exprToTrans: ${unpackedString(baseExpr, model)}")
+            for ((name, expr) in printVars) {
+                val ufp = transformer.mapFpToUnpackedFp[expr.decl]
+                val evalUnpacked = unpackedString(ufp ?: expr, model)
+                println("$name :: $evalUnpacked")
+            }
+        }
+        assertNotEquals(KSolverStatus.SAT, status)
```

Comment:
It is better check equals with unsat. And manually ignore tests with unknowns. Currently, tests with unknowns are marked as passed.